pandas/tests/strings/test_get_dummies.py

```diff
@@ -32,22 +32,94 @@ def test_get_dummies_index():
     tm.assert_index_equal(result, expected)
 
 
-def test_get_dummies_with_name_dummy(any_string_dtype):
-    # GH 12180
-    # Dummies named 'name' should work as expected
-    s = Series(["a", "b,name", "b"], dtype=any_string_dtype)
-    result = s.str.get_dummies(",")
-    expected = DataFrame([[1, 0, 0], [0, 1, 1], [0, 1, 0]], columns=["a", "b", "name"])
+def test_get_dummies_int8_dtype():
```

Comment:
Can you parametrize these tests with `dtype`.

Reply:
In addition, can you add `dtype=str`, PyArrow, and nullable dtypes (e.g. "Int64"). Specifying PyArrow and nullable dtypes currently fails:

```python
ser = pd.Series(["a|b", "a", "a|c"], dtype="string[pyarrow]")
ser.str.get_dummies(dtype=pd.ArrowDtype(pa.int64()))
```

but is successful with `pd.get_dummies`

```python
pd.get_dummies(ser, dtype=pd.ArrowDtype(pa.int64()))
```

I think this will need to be fixed. You may find it necessary to have multiple tests - perhaps one for NumPy (which are already present), one for `str`, one for PyArrow etc. But just try to consolidate with `pytest.parametrize` as much as is reasonable.